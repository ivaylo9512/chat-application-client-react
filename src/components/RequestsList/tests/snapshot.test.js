import * as redux from 'react-redux';
import { shallow } from 'enzyme';
import RequestList from 'components/RequestsList/RequestsList';

describe('RequestList snapshot tests', () => {
    beforeEach(() => {
        jest.spyOn(redux, 'useDispatch').mockReturnValue(jest.fn());
    })
    
    const createWrapper = (state) => {
        jest.spyOn(redux, 'useSelector').mockReturnValue(state);

        return shallow(
            <RequestList />
        )
    }

    it('should render info requests data is empty', () => {
        const wrapper = createWrapper([]);

        expect(wrapper).toMatchSnapshot();
    })

    it('should render list when requests data is not empty', () => {
        const wrapper = createWrapper([{ id: 5, createdAt: '2021-08-23', sender: { firstName: 'firstname', lastName: 'lastName', profileImage: 'image.png' }}, 
            { id: 6, createdAt: '2021-08-23', sender: { firstName: 'firstname2', lastName: 'lastName2', profileImage: 'image2.png' }}]);
        
        expect(wrapper).toMatchSnapshot();
    })

    it('should render empty Container when requests returns undefined', () => {
        const wrapper = createWrapper();

        expect(wrapper).toMatchSnapshot();
    })
})
