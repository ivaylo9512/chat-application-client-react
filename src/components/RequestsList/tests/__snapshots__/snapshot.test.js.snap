// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestList snapshot tests should render empty Container when requests returns undefined 1`] = `<styled.div />`;

exports[`RequestList snapshot tests should render info requests data is empty 1`] = `
<styled.div>
  <span
    data-testid="info"
  >
    <b>
      No requests found.
    </b>
  </span>
</styled.div>
`;

exports[`RequestList snapshot tests should render list when requests data is not empty 1`] = `
<styled.div>
  <Memo()
    key="5"
    request={
      Object {
        "createdAt": "2021-08-23",
        "id": 5,
        "sender": Object {
          "firstName": "firstname",
          "lastName": "lastName",
          "profileImage": "image.png",
        },
      }
    }
  />
  <Memo()
    key="6"
    request={
      Object {
        "createdAt": "2021-08-23",
        "id": 6,
        "sender": Object {
          "firstName": "firstname2",
          "lastName": "lastName2",
          "profileImage": "image2.png",
        },
      }
    }
  />
</styled.div>
`;
